File: frontend/src/types/apikey.ts
export type CreateApiKeyResult = {
  ok: boolean
  apiKey: ApiKey
  rawKey: string
  code?: string
  message?: string
}

export type ApiKey = {
  name: string
  isRevoked: boolean
  createdAt: string
  expiresAt: string
  publicId: string
}

export type ApiKeyPageContentProps = {
  readonly isGitHubAuthEnabled: boolean
}

File: backend/apps/nest/graphql/nodes/api_key.py
"""GraphQL node for APIKey model."""

import strawberry_django

from apps.nest.models import ApiKey


@strawberry_django.type(
    ApiKey,
    fields=[
        "created_at",
        "is_revoked",
        "expires_at",
        "name",
        "public_id",
    ],
)
class ApiKeyNode:
    """GraphQL node for API keys."""
